src/extensions/core/electronTasks.ts

```diff
@@ -0,0 +1,154 @@
+        'Deletes the desktop app config and load the welcome screen.',
+      detail:
+        'Delete the desktop app config, restart the app, and load the installation screen.',
+      confirmText: 'Delete all saved config and reinstall?',
+      button: {
+        icon: PrimeIcons.EXCLAMATION_TRIANGLE,
+        text: 'Reinstall'
+      }
+    },
+    {
+      id: 'pythonPackages',
+      requireConfirm: true,
+      execute: async () => {
+        try {
+          await electron.uv.installRequirements()
+          return true
+        } catch (error) {
+          return false
+        }
+      },
+      name: 'Install python packages',
+      description: 'Installs the base python packages required to run ComfyUI.',
+      errorDescription:
+        'Python packages that are required to run ComfyUI are not installed.',
+      detail:
+        'This will install the python packages required to run ComfyUI. This includes torch, torchvision, and other dependencies.',
+      usesTerminal: true,
+      isInstallationFix: true,
+      button: {
+        icon: PrimeIcons.DOWNLOAD,
+        text: 'Install'
+      }
+    },
+    {
+      id: 'uv',
+      execute: () =>
+        openUrl('https://docs.astral.sh/uv/getting-started/installation/'),
+      name: 'uv executable',
+      description: 'uv installs and maintains the python environment.',
+      detail:
+        "This will open the download page for Astral's uv tool. uv is used to install python and manage python packages.",
+      button: {
+        icon: 'pi pi-asterisk',
+        text: 'Download'
+      }
+    },
+    {
+      id: 'uvCache',
+      severity: 'danger',
+      requireConfirm: true,
+      execute: async () => await electron.uv.clearCache(),
+      name: 'uv cache',
+      description: 'Remove the Astral uv cache of python packages.',
+      detail:
+        'This will remove the uv cache directory and its contents. All downloaded python packages will need to be downloaded again.',
+      confirmText: 'Delete uv cache of python packages?',
+      isInstallationFix: true,
+      button: {
+        icon: PrimeIcons.TRASH,
+        text: 'Clear cache'
+      }
+    },
+    {
+      id: 'venvDirectory',
+      severity: 'danger',
+      requireConfirm: true,
+      execute: async () => await electron.uv.resetVenv(),
+      name: 'Reset virtual environment',
+      description:
+        'Remove and recreate the .venv directory. This removes all python packages.',
+      detail:
+        'The python environment is where ComfyUI installs python and python packages. It is used to run the ComfyUI server.',
+      confirmText: 'Delete the .venv directory?',
+      usesTerminal: true,
+      isInstallationFix: true,
+      button: {
+        icon: PrimeIcons.FOLDER,
+        text: 'Recreate'
+      }
+    }
+  ]
+
+  // Use a minimum run time to ensure tasks "feel" like they have run
+  for (const task of tasks) {
+    task.loading = minDurationRef(true, 250)
+  }
+
+  return tasks
```

Comment:
Let's split the state tracking and task declaration. Moving these task states to a pinia store instead. Currently if this hook is called, new copies of tasks are created, and their states are not shared.

I think in this case specifc case for maintenance tasks, we want them to be global singleton instead of create copies at each call.